File: examples/developer-hub-python/make_query_coston2.py
import asyncio
import json

import aiohttp
from web3 import AsyncHTTPProvider, AsyncWeb3
from web3.middleware.geth_poa import async_geth_poa_middleware


async def main() -> str:
    registry_addr = "0xaD67FE66660Fb8dFE9d6b1b4240d8650e30F6019"
    w3 = AsyncWeb3(
        AsyncHTTPProvider("https://coston2-api.flare.network/ext/C/rpc"),
        middlewares=[async_geth_poa_middleware],
    )
    params = {
        "module": "contract",
        "action": "getabi",
        "address": registry_addr,
    }
    async with (
        aiohttp.ClientSession() as session,
        session.get(
            "https://coston2-explorer.flare.network/api", params=params
        ) as response,
    ):
        res = await response.json()
        abi = json.loads(res["result"])
    registry = w3.eth.contract(address=w3.to_checksum_address(registry_addr), abi=abi)
    res = await registry.functions.getContractAddressByName("WNat").call()
    print(f"WNat address: {res}")
    # WNat address: 0xC67DCE33D7A8efA5FfEB961899C73fe01bCe9273
    return res


if __name__ == "__main__":
    asyncio.run(main())

File: examples/developer-hub-javascript/make_query_coston2.js
import { Web3 } from "web3";

const registry_addr = "0xaD67FE66660Fb8dFE9d6b1b4240d8650e30F6019";
const base_url = "https://coston2-explorer.flare.network/api";

export async function main() {
  // Fetch ABI
  const params = `?module=contract&action=getabi&address=${registry_addr}`;
  const response = await fetch(base_url + params);
  const abi = JSON.parse((await response.json())["result"]);

  // Query contract
  const w3 = new Web3("https://coston2-api.flare.network/ext/C/rpc");
  const registry = new w3.eth.Contract(abi, registry_addr);
  const res = await registry.methods.getContractAddressByName("WNat").call();
  console.log("WNat address: ", res);
  // WNat address: 0xC67DCE33D7A8efA5FfEB961899C73fe01bCe9273

  return res;
}
